invoice_currency_validation/models/account_invoice.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# © 2017 Savoir-faire Linux
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = 'account.invoice'
+
+    @api.onchange('currency_id')
+    def _onchange_currency_id(self):
+        if self.journal_id.currency_id != self.currency_id:
+            journal_type = 'sale' if self.type == 'out_invoice' else 'purchase'
+
+            journal_id = self.env['account.journal'].search([
+                ('type', '=', journal_type),
+                ('currency_id', '=', self.currency_id.id)
```

Comment:
The currency is the currency of the company. the expression should be 
`'|', ('currency_id', '=', self.currency_id.id), ('currency_id', '=', False)`